dask_ml/decomposition/incremental_pca.py

```diff
@@ -0,0 +1,340 @@
+                             "the batch number of samples "
+                             "%d." % (self.n_components, n_samples))
+        else:
+            self.n_components_ = self.n_components
+
+        if (self.components_ is not None) and (self.components_.shape[0] !=
+                                               self.n_components_):
+            raise ValueError("Number of input features has changed from %i "
+                             "to %i between calls to partial_fit! Try "
+                             "setting n_components to a fixed value." %
+                             (self.components_.shape[0], self.n_components_))
+
+        # This is the first partial_fit
+        if not hasattr(self, 'n_samples_seen_'):
+            self.n_samples_seen_ = 0
+            self.mean_ = .0
+            self.var_ = .0
+        
+        # Update stats - they are 0 if this is the first step
+        last_sample_count = np.tile(np.expand_dims(self.n_samples_seen_, 0), X.shape[1])
+        col_mean, col_var, n_total_samples = \
+            _incremental_mean_and_var(
+                X, last_mean=self.mean_, last_variance=self.var_,
+                last_sample_count=last_sample_count)
+        n_total_samples = n_total_samples[0]
+        
+        # Whitening
+        if self.n_samples_seen_ == 0:
+            # If it is the first step, simply whiten X
+            X -= col_mean
+        else:
+            col_batch_mean = np.mean(X, axis=0)
+            X -= col_batch_mean
+            # Build matrix of combined previous basis and new data
+            mean_correction = \
+                np.sqrt((self.n_samples_seen_ * n_samples) /
+                        n_total_samples) * (self.mean_ - col_batch_mean)
+            X = np.vstack((self.singular_values_.reshape((-1, 1)) *
+                           self.components_, X, mean_correction))
+
+        solver = self._get_solver(X, self.n_components_)
+        if solver in {"full", "tsqr"}:
+            if hasattr(X, 'rechunk'):
+                X = da.rechunk(X, (X.chunks[0], -1))
+            U, S, V = linalg.svd(X)
+        else:
+            # randomized
+            random_state = check_random_state(self.random_state)
+            seed = draw_seed(random_state, np.iinfo("int32").max)
+            n_power_iter = self.iterated_power
+            U, S, V = linalg.svd_compressed(
+                X, self.n_components, n_power_iter=n_power_iter, seed=seed
+            )
+        U, V = svd_flip(U, V)
+        explained_variance = S ** 2 / (n_total_samples - 1)
+        explained_variance_ratio = S ** 2 / np.sum(col_var * n_total_samples)
+
+        self.n_samples_seen_ = n_total_samples
+        self.components_ = V[:self.n_components_]
+        self.singular_values_ = S[:self.n_components_]
+        self.mean_ = col_mean
+        self.var_ = col_var
+        self.explained_variance_ = explained_variance[:self.n_components_]
```

Comment:
Can you check the types of all these vs. those in `dask_ml.decomposition.PCA`? I notice some that differ.

Which do you find more useful here: a Dask Array or a concrete ndarray?